### ISSUE TITLE
Example code doesn't work

### ISSUE THREAD

**Issue by Ascenio** [rikulo/socket.io-client-dart#134]
It doesn't even compile. Also, why is `socket_io` a dependency?

**Comment by jumperchen** [comment]
FYI: https://github.com/rikulo/socket.io-client-dart/tree/master/test

**Comment by Ascenio** [comment]
Consider the example code (taken from README): 
```dart
import 'package:socket_io/socket_io.dart';
import 'package:socket_io_client/socket_io_client.dart' as IO;

void main() {
  var io = Server();
  io.on('connection', (Socket client) {
    print('connection default namespace');
    client.on('msg', (data) {
      print('data from default => $data');
      client.emit('fromServer', "ok");
    });
  });
  io.listen(3000);
  // Dart client
  IO.Socket socket = IO.io('http://localhost:3000');
  socket.onConnect((_) {
    print('connect');
    socket.emit('msg', 'test');
  });
  socket.on('event', (data) => print(data));
  socket.onDisconnect((_) => print('disconnect'));
  socket.on('fromServer', (_) => print(_));
}
```
Well, I'm right, the example code doesn't work. First it gives on:
```
type '(Socket) => Null' is not a subtype of type '(dynamic) => dynamic'
```
So if you make `Socket client` be just `client` it arraises another error:
```
2020-12-21 16:05:50.686054:stream:0
INFO: Rikulo Stream Server 2.6.0 starting on 0.0.0.0:3000
Home: /home/ascenio/code/sqlite_observer/sqlite_observer/lib/src
connection default namespace
2020-12-21 16:05:50.797373:stream:1
SHOUT: 2020-12-21 16:05:50.796992:Bad state: Header already sent
```
But as soon as I change the client as follows it just works:
```dart
IO.Socket socket = IO.io('http://localhost:3000', <String, dynamic>{
  'transports': ['websocket'],
});
```
But then if `transports` is a mandatory header, why isn't default? Or did I get it wrong?
Also, if thats the case maybe should update the example code to contain the header.

**Comment by jumperchen** [comment]
```
IO.Socket socket = IO.io('http://localhost:3000', <String, dynamic>{
  'transports': ['websocket'],
});
```
TL;DR
By default, the `transports` is  `['polling', 'websocket']`, but the 'polling' does not support in Flutter; this library was designed for dart web at beginning, but dart web is not popular than Flutter, so you have to specify 'transports': ['websocket'] when you are using in Flutter env., indeed Dart also provides a way to runtime check whether the env. is running in Flutter or Dart web, but since Flutter Web also is supported for now, I don't know whether there's a flag to check such env. at runtime.
